src/V3EmitCImp.cpp

```diff
@@ -425,7 +425,7 @@ class EmitCImp final : EmitCFunc {
     }
     // Predicate to check if we actually need to emit anything into the common implementation file.
     // Used to avoid creating empty output files.
-    bool hasCommonImp(const AstNodeModule* modp) const {
+    bool hasCommonImp(const AstNodeModule* modp) const VL_MT_SAFE {
```

Comment:
This is safe only if assuming some other visitor isn't changing the tree topology.  I think we need e.g. VL_MT_STABLE_TREE or something to state that assumption.